### ISSUE TITLE
'str' object has no attribute 'supported' when building documentation using m2r2

### ISSUE THREAD

**Issue by brsc2909** [sphinx-doc/sphinx#8395]
**Describe the bug**
when i add the m2r2 extension to my project is get the error below

**To Reproduce**
```bash
$ pip install m2r2
$ mkdir testdoc && cd testdoc
```
Add m2r2 to extensions in conf.py
```python
# Configuration file for the Sphinx documentation builder.
#
# This file only contains a selection of the most common options. For a full
# list see the documentation:
# https://www.sphinx-doc.org/en/master/usage/configuration.html

# -- Path setup --------------------------------------------------------------

# If extensions (or modules to document with autodoc) are in another directory,
# add these directories to sys.path here. If the directory is relative to the
# documentation root, use os.path.abspath to make it absolute, like shown here.
#
# import os
# import sys
# sys.path.insert(0, os.path.abspath('.'))


# -- Project information -----------------------------------------------------

project = 'test'
copyright = '2020, bs'
author = 'bs'


# -- General configuration ---------------------------------------------------

# Add any Sphinx extension module names here, as strings. They can be
# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
# ones.
extensions = ['m2r2']

# Add any paths that contain templates here, relative to this directory.
templates_path = ['_templates']

# List of patterns, relative to source directory, that match files and
# directories to ignore when looking for source files.
# This pattern also affects html_static_path and html_extra_path.
exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']


# -- Options for HTML output -------------------------------------------------

# The theme to use for HTML and HTML Help pages.  See the documentation for
# a list of builtin themes.
#
html_theme = 'alabaster'

# Add any paths that contain custom static files (such as style sheets) here,
# relative to this directory. They are copied after the builtin static files,
# so a file named "default.css" will overwrite the builtin "default.css".
html_static_path = ['_static']

```
Make docs
```bash
$ make html
```
Error:
```
$ make html
Running Sphinx v3.3.0

Exception occurred:
  File "/home/username/PycharmProjects/project/venv/lib/python3.8/site-packages/sphinx/registry.py", line 266, in add_source_parser
    for filetype in parser.supported:
AttributeError: 'str' object has no attribute 'supported'
The full traceback has been saved in /tmp/sphinx-err-bk5vovm_.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:20: html] Error 2
```
**Expected behavior**

**Environment info**
- OS: Linux manjaro 20.1 KDE plasma
- Python version:  3.8.6
- Sphinx version: 3.3.0
- Sphinx extensions:  m2r2

**Comment by brsc2909** [comment]
This can be closed it's fixed in 
https://github.com/CrossNox/m2r2/pull/14/commits/484b792ffded6a0783a89c01dbff40ba53ffb541

**Comment by cglacet** [comment]
I'm still getting this error with the latest version of m2r2 `0.2.5` (and sphinx `3.3.0`).

**Comment by cglacet** [comment]
On the other hand the proposed patch does work.

**Comment by brsc2909** [comment]
> I'm still getting this error with the latest version of m2r2 `0.2.5` (and sphinx `3.3.0`).

The fix hasn't been merged yet, it's still waiting on this pull request https://github.com/CrossNox/m2r2/pull/14.

**Comment by cglacet** [comment]
"8 days ago" ok. Is there a single person with merge rights on m2r2? It's a bit strange to spend 8 days approving this kind of edit. Anyway, thanks for the issue you saved me quite a bit of time :D

**Comment by cglacet** [comment]
In the mean time, `conf.py`:

```python
"""Patching m2r2"""
import m2r2

current_m2r2_setup = m2r2.setup

def patched_m2r2_setup(app):
    try:
        return current_m2r2_setup(app)
    except (AttributeError):
        app.add_source_suffix(".md", "markdown")
        app.add_source_parser(m2r2.M2RParser)
    return dict(
        version=m2r2.__version__, parallel_read_safe=True, parallel_write_safe=True,
    )

m2r2.setup = patched_m2r2_setup
```

**Comment by Sherryzhang2** [comment]
I also met this problem. Has this issue been fixed and merged into any release version?

**Comment by KaiSchwarz-cnic** [comment]
I am also running into this problem. But will try to get the above workaround applied.

**Comment by KaiSchwarz-cnic** [comment]
fyi I am reading somewhere else about a plugin m2r2 - looks like m2r is abandoned. replacing m2r might fix it.
https://github.com/CrossNox/m2r2/issues/13

I'll give that a try and will post the result. -> Yes, fixed.

**Comment by martinm82** [comment]
@papakai using m2r2 fixes indeed the problem. Thanks for the workaround
